core-httpclient-impl/src/main/java/com/optimizely/ab/OptimizelyFactory.java

```diff
@@ -279,23 +279,37 @@ public static Optimizely newDefaultInstance(String sdkKey, String fallback, Stri
      * @param customHttpClient  Customizable CloseableHttpClient to build OptimizelyHttpClient.
      * @return A new Optimizely instance
      */
-    public static Optimizely newDefaultInstance(String sdkKey, String fallback, String datafileAccessToken, CloseableHttpClient customHttpClient) {
+    public static Optimizely newDefaultInstance(
+        String sdkKey,
+        String fallback,
+        String datafileAccessToken,
+        CloseableHttpClient customHttpClient
+    ) {
+        OptimizelyHttpClient optimizelyHttpClient = customHttpClient == null ? null : new OptimizelyHttpClient(customHttpClient);
```

Comment:
Just want to know, if CustomHttpClient is not provided, then how OptimizelyHttpClient get initialized?